codes/c/chapter_dynamic_programming/edit_distance.c

```diff
@@ -0,0 +1,147 @@
+/**
+ * File: edit_distance.c
+ * Created Time: 2023-10-02
+ * Author: Zuoxun (845242523@qq.com)
+ */
+
+#include "../utils/common.h"
+
+int n, m;
+int s_length, t_length;
+
+/* 编辑距离：暴力搜索 */
+int editDistanceDFS(char *s, char *t, int i, int j) {
+    // 若 s 和 t 都为空，则返回 0
+    if (i == 0 && j == 0)
+        return 0;
+    // 若 s 为空，则返回 t 长度
+    if (i == 0)
+        return j;
+    // 若 t 为空，则返回 s 长度
+    if (j == 0)
+        return i;
+    // 若两字符相等，则直接跳过此两字符
+    if (s[i - 1] == t[j - 1])
+        return editDistanceDFS(s, t, i - 1, j - 1);
+    // 最少编辑步数 = 插入、删除、替换这三种操作的最少编辑步数 + 1
+    int insert = editDistanceDFS(s, t, i, j - 1);
+    int del = editDistanceDFS(s, t, i - 1, j);
+    int replace = editDistanceDFS(s, t, i - 1, j - 1);
+    // 返回最少编辑步数
+    return min(min(insert, del), replace) + 1;
+}
+
+/* 编辑距离：记忆化搜索 */
+int editDistanceDFSMem(char *s, char *t, int mem[][m + 1], int i, int j) {
+    // 若 s 和 t 都为空，则返回 0
+    if (i == 0 && j == 0)
+        return 0;
+    // 若 s 为空，则返回 t 长度
+    if (i == 0)
+        return j;
+    // 若 t 为空，则返回 s 长度
+    if (j == 0)
+        return i;
+    // 若已有记录，则直接返回之
+    if (mem[i][j] != -1)
+        return mem[i][j];
+    // 若两字符相等，则直接跳过此两字符
+    if (s[i - 1] == t[j - 1])
+        return editDistanceDFSMem(s, t, mem, i - 1, j - 1);
+    // 最少编辑步数 = 插入、删除、替换这三种操作的最少编辑步数 + 1
+    int insert = editDistanceDFSMem(s, t, mem, i, j - 1);
+    int del = editDistanceDFSMem(s, t, mem, i - 1, j);
+    int replace = editDistanceDFSMem(s, t, mem, i - 1, j - 1);
+    // 记录并返回最少编辑步数
+    mem[i][j] = min(min(insert, del), replace) + 1;
+    return mem[i][j];
+}
+
+/* 编辑距离：动态规划 */
+int editDistanceDP(char *s, char *t) {
```

Comment:
It'd be clearer to add the function parameters `n` , `m` rather than define them out of the function.

Reply:
Thx for the explaination. Pending for reviewing by @Gonglja.

Reply:
I've edited it according to your suggestion, added parameter n, m to the function, but reserved outside parameter m in order to solve the problem mentioned above.
You can compare these two writing methods and choose one.